### ISSUE TITLE
Bind topology t0 to VMs. base vm = VM0100 failed

### ISSUE THREAD

**Issue by lguohan** [Azure/sonic-mgmt#1250]
**Description**

**Steps to reproduce the issue:**
1. refresh-dut
2.
3.

**Describe the results you received:**

```
TASK [vm_set : Bind topology t0 to VMs. base vm = VM0100] **********************
Monday 02 December 2019  04:50:22 +0000 (0:00:00.051)       0:03:56.997 ******* 
fatal: [STR-ACS-VSERV-01]: FAILED! => {"changed": false, "msg": "ret_code=1, error message=eth20: error fetching interface information: Device not found\n. cmd=ifconfig -a"}
```

```
lgh@jenkins-worker-12:/tmp$ cat vmtopology.exception.txt 
Traceback (most recent call last):
  File "/tmp/ansible_vm_topology_payload_HGgiq8/__main__.py", line 705, in main
    net.add_veth_ports_to_docker()
  File "/tmp/ansible_vm_topology_payload_HGgiq8/__main__.py", line 218, in add_veth_ports_to_docker
    self.add_veth_if_to_docker(ext_if, int_if)
  File "/tmp/ansible_vm_topology_payload_HGgiq8/__main__.py", line 308, in add_veth_if_to_docker
    self.update()
  File "/tmp/ansible_vm_topology_payload_HGgiq8/__main__.py", line 153, in update
    self.host_ifaces = VMTopology.ifconfig('ifconfig -a')
  File "/tmp/ansible_vm_topology_payload_HGgiq8/__main__.py", line 537, in ifconfig
    out = VMTopology.cmd(cmdline)
  File "/tmp/ansible_vm_topology_payload_HGgiq8/__main__.py", line 499, in cmd
    raise Exception("ret_code=%d, error message=%s. cmd=%s" % (ret_code, stderr, cmdline))
Exception: ret_code=1, error message=eth20: error fetching interface information: Device not found
. cmd=ifconfig -a
```

some race condition when there are kernel interface created/deleted to cause ifconfig -a to fail. Need to add retry in vm_topology.py.

**Describe the results you expected:**

**Additional information you deem important:**

    **Output of `show version`:**

    ```
    (paste your output here)
    ```

    **Attach debug file `sudo generate_dump`:**

    ```
    (paste your output here)
    ```

**Comment by lguohan** [comment]
addressed in #1256

**Comment by deba-arista** [comment]
I am facing a similar issue.... Please check it!
TASK [vm_set : Bind topology t0 to VMs. base vm = VM0100] *******************************************************************************
task path: /var/AzDevOps/sonic-mgmt/ansible/roles/vm_set/tasks/add_topo.yml:200
Wednesday 25 October 2023  13:24:28 +0000 (0:00:07.402)       0:02:57.707 *****
fatal: [STR-ACS-SERV-01]: FAILED! => {
    "msg": "The task includes an option with an undefined variable. The error was: 'ptf_extra_mgmt_ip' is undefined\n\nThe error appears to be in '/var/AzDevOps/sonic-mgmt/ansible/roles/vm_set/tasks/add_topo.yml': line 200, column 5, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n  - name: Bind topology {{ topo }} to VMs. base vm = {{ VM_base }}\n    ^ here\nWe could be wrong, but this one looks like it might be an issue with\nmissing quotes. Always quote template expression brackets when they\nstart a value. For instance:\n\n    with_items:\n      - {{ foo }}\n\nShould be written as:\n\n    with_items:\n      - \"{{ foo }}\"\n"
}

PLAY RECAP ******************************************************************************************************************************
STR-ACS-SERV-01            : ok=225  changed=17   unreachable=0    failed=1    skipped=226  rescued=0    ignored=0
